File: src/main/php/Easy/BackspaceCompare.php
<?php

declare(strict_types=1);

namespace Src\Easy;

class BackspaceCompare
{
    public function process(string $text): string
    {
        $result = [];
        for ($i = 0; $i < strlen($text); $i++) {
            if ($text[$i] === '#') {
                if (! empty($result)) {
                    array_pop($result);
                }

            } else {
                $result[] = $text[$i];
            }
        }

        return implode('', $result);
    }

    public function solve(string $s, string $t): bool
    {
        return $this->process($s) === $this->process($t);
    }
}

File: src/main/cpp/easy/backspace_compare.cpp
#include "backspace_compare.h"

std::string processString(std::string text) {
  std::vector<char> stack;
  for (char c : text) {
   if (c == '#') {
            if (!stack.empty()) {
                stack.pop_back();
            }
        }else {
      stack.push_back(c);
    }
  }
  return std::string(stack.begin(), stack.end());
}
bool backspaceCompare(std::string s, std::string t) {
  return processString(s) == processString(t);
}
